File: dongyimai-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function ($http) {

    this.submitOrder = function (order) {
        return $http.post('../order/add.do',order)
    }

    this.findCartList = function () {
        return $http.get('../cart/findCartList.do');
    }
    this.add = function (itemId,num) {
        return $http.post('../cart/addGoodsToCartList.do?itemId='+itemId + '&num='+num);
    }
})

File: dongyimai-cart-interface/src/main/java/com/dongyimai/cart/service/CartService.java
package com.dongyimai.cart.service;

import com.dongyimai.group.Cart;

import java.util.List;

public interface CartService {

    /**
     * 购物车合并
     *
     */
    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);
    /**
     * 添加购物车商品
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);
    /**
     * 保存redis中商品
     */
    public void saveRedisCartList(List<Cart> cartList,String username);

    /**
     * 从redis中获取商品信息
     * @return
     */
    public List<Cart> findRedisCartList(String username);
}
